### PR TITLE
fix: designated screen load on route

### PR DESCRIPTION
**Pull request by sagardspeed2**
## Describe your changes
fix designated screen load on route

## Issue ticket number and link

Closes #

## Checklist before requesting a review

- [x] I have performed a self-review of my code
- [ ] I have added tests
- [x] I have added a changeset `pnpm changeset add`
- [x] I have added example usage in the kitchen sink app

### CODE REVIEW

File: packages/runtime/src/runtime/entry.tsx [hunk 1]
```diff
@@ -20,24 +21,32 @@ export const EnsembleEntry: React.FC<EnsembleEntryProps> = ({
...
-    if (!(hasMenu && location.pathname === "/")) {
+    const navigateToPath = (pathname: string, search = ""): void => {
+      navigate({ pathname, search });
+    };
+
+    const getLowerCasePath = (name: string): string => `/${name.toLowerCase()}`;
+
+    const currentPath = location.pathname;
+
+    if (screen && currentPath !== getLowerCasePath(screen.name)) {
```
Comment (evshi) [inline packages/runtime/src/runtime/entry.tsx:36]: This is my poor naming, can we rename this to initialScreen?

File: packages/runtime/src/runtime/entry.tsx [hunk 2]
```diff
@@ -20,24 +21,32 @@ export const EnsembleEntry: React.FC<EnsembleEntryProps> = ({
...
-    });
+    if (hasMenu) {
+      if (currentPath === "/") {
+        const selectedItem =
+          entry.items.find((item) => item.selected) ?? entry.items[0];
+        navigateToPath(getLowerCasePath(selectedItem.page), location.search);
+      }
+    } else if (currentPath === "/") {
+      navigateToPath(getLowerCasePath(entry.name));
+    }
```
Comment (evshi) [inline packages/runtime/src/runtime/entry.tsx:49]: I think you keep the check for selectedItem && menu, and then lastly navigate to home...so the priority should be like

if (path is not root)

if (screen)

if (hasMenu && selectedItem)

goHome

it's not really elseif it's just a series of conditions and can return early if met.

File: packages/runtime/src/runtime/entry.tsx [hunk 3]
```diff
@@ -20,24 +21,35 @@ export const EnsembleEntry: React.FC<EnsembleEntryProps> = ({
...
+    };
+
+    const getLowerCasePath = (name: string): string => `/${name.toLowerCase()}`;
+
+    const initialScreen = location.pathname;
+
+    if (initialScreen !== "/") {
+      return;
+    }
+
```
Comment (evshi) [inline packages/runtime/src/runtime/entry.tsx:39]: This is incorrect. We should navigate to the path specified in the URL

**Review by evshi — CHANGES_REQUESTED** [review]

File: packages/runtime/src/runtime/entry.tsx (hunk 3 above)
Comment (sagardspeed2) [reply packages/runtime/src/runtime/entry.tsx:39]: You mean `location.pathname` ?

File: packages/runtime/src/runtime/entry.tsx [hunk 4]
```diff
@@ -20,24 +21,35 @@ export const EnsembleEntry: React.FC<EnsembleEntryProps> = ({
...
-      navigate({
-        pathname: `/${screen.name.toLowerCase()}`,
-        search: location.search,
-      });
+    if (!keys(entry).length) {
       return;
     }
 
-    if (!(hasMenu && location.pathname === "/")) {
+    const navigateToPath = (pathname: string, search = ""): void => {
```
Comment (evshi) [inline packages/runtime/src/runtime/entry.tsx:28]: Can we remove this? it isn't doing much...

File: packages/runtime/src/runtime/entry.tsx [hunk 5]
```diff
@@ -20,24 +21,35 @@ export const EnsembleEntry: React.FC<EnsembleEntryProps> = ({
...
+      navigate({ pathname, search });
+    };
+
+    const getLowerCasePath = (name: string): string => `/${name.toLowerCase()}`;
```
Comment (evshi) [inline packages/runtime/src/runtime/entry.tsx:32]: We probably don't need this - you can just store this in a variable.

File: packages/runtime/src/runtime/entry.tsx (hunk 3 above)
Comment (evshi) [reply packages/runtime/src/runtime/entry.tsx:39]: Can you leave a comment why this works?

File: packages/runtime/src/runtime/entry.tsx [hunk 6]
```diff
@@ -20,24 +21,35 @@ export const EnsembleEntry: React.FC<EnsembleEntryProps> = ({
...
+
+    const initialScreen = location.pathname;
```
Comment (evshi) [inline packages/runtime/src/runtime/entry.tsx:34]: Rename `initialScreen` to `path`
Rename `screen` to `initialScreen`

File: packages/runtime/src/runtime/entry.tsx [hunk 7]
```diff
@@ -20,24 +21,35 @@ export const EnsembleEntry: React.FC<EnsembleEntryProps> = ({
...
       return;
     }
 
-    const selectedItem =
-      entry.items.find((item) => item.selected) ?? entry.items[0];
-    navigate({
-      pathname: `/${selectedItem.page.toLowerCase()}`,
-      search: location.search,
-    });
+    navigateToPath(getLowerCasePath(entry.name));
```
Comment (evshi) [inline packages/runtime/src/runtime/entry.tsx:52]: Should include search here?

File: packages/runtime/src/runtime/entry.tsx [hunk 8]
```diff
@@ -20,24 +21,35 @@ export const EnsembleEntry: React.FC<EnsembleEntryProps> = ({
 
   const hasMenu = "items" in entry;
   useEffect(() => {
-    if (screen && location.pathname !== `/${screen.name.toLowerCase()}`) {
-      navigate({
-        pathname: `/${screen.name.toLowerCase()}`,
-        search: location.search,
-      });
+    if (!keys(entry).length) {
```
Comment (evshi) [inline packages/runtime/src/runtime/entry.tsx:24]: Why is this check needed?

**Review by evshi — APPROVED** [review]
Can you please sanity check atlas app as well?

**Pull request merged**